node/clients/profile/profileV2.ts

```diff
@@ -0,0 +1,303 @@
+
+    if (!(profile as Profile)) {
+      const profileCast = profile as Profile
+      profileCast.gender = profileCast.gender || ""
+      profileCast.document = profileCast.document || ""
+    }
+
+    return this.patch(
+      `${this.baseUrl}/${userKey}?an=${this.account}&alternativeKey=${alternativeKey}`,
+      profile,
+      {
+        metric: 'profile-system-v2-updateProfileInfo',
+        params: {
+          extraFields: customFields,
+        },
+      }
+    )
+  }
+
+  public getUserAddresses = (_: CurrentProfile, currentUserProfile: Profile, piiRequest?: PIIRequest) => {
+    let url = this.getPIIUrl(`${this.baseUrl}/${currentUserProfile.id}/addresses`, undefined, piiRequest)
+
+    return this.get<Address[]>(url, { metric: 'profile-system-v2-getUserAddresses', })
+      .then((addresses: AddressV2[]) => this.translateToV1Address(addresses))
+      .catch<any>(e => {
+        const { response } = e as AxiosError
+        const { status } = response!
+        if (status == 404) {
+          return [] as AddressV2[]
+        }
+
+        return statusToError(e)
+      })
+  }
+
+  private translateToV1Address = (addresses: AddressV2[]) =>
+    addresses.map((address: AddressV2) => {
+      let addressV2 = address.document
+
+      return {
+        addressName: addressV2.name,
+        city: addressV2.localityAreaLevel1,
+        complement: addressV2.extend,
+        country: addressV2.countryCode,
+        geoCoordinates: addressV2.geoCoordinates,
+        id: address.id,
+        number: addressV2.streetNumber,
+        postalCode: addressV2.postalCode,
+        receiverName: addressV2.receiverName,
+        reference: addressV2.nearly,
+        state: addressV2.administrativeAreaLevel1,
+        street: addressV2.route,
+        userId: addressV2.profileId,
+        addressType: addressV2.addressType || "residential",
+        neighborhood: addressV2.neighborhood,
+      } as Address
+    })
+
+  private translateToV2Address = (addresses: Address[]) =>
+    addresses.map((address: Address) => {
+      return {
+        id: address.id,
+        document: {
+          administrativeAreaLevel1: address.state,
+          addressType: address.addressType || "residential",
+          countryCode: address.country,
+          extend: address.complement || "",
+          geoCoordinates: address.geoCoordinates,
+          localityAreaLevel1: address.city,
+          name: address.addressName,
+          nearly: address.reference || "",
+          postalCode: address.postalCode,
+          profileId: address.userId,
+          route: address.street,
+          streetNumber: address.number,
+          receiverName: address.receiverName,
+          neighborhood: address.neighborhood,
+        }
+      } as AddressV2
+  })
+
+  private mapAddressesObjToList(addressesObj: any): Address[] {
+    return Object.entries<string>(addressesObj).map(
+      ([key, stringifiedObj]) => {
+        let address = JSON.parse(stringifiedObj) as Address
```

Comment:
```suggestion
        const address = JSON.parse(stringifiedObj) as Address
```
Would be good to wrap it json.parse with a try catch block